packages/synapse-react-client/src/components/TableColumnSchemaEditor/TableColumnSchemaEditor.tsx

```diff
@@ -81,32 +50,35 @@ export default function TableColumnSchemaEditor(
       },
     )
 
-  const { mutate, isLoading: isMutating } = useUpdateTable()
+  const { mutate, isLoading: isMutating, error } = useUpdateTableColumns()
 
-  const isLoading = isLoadingEntity || isLoadingColumnModels || isMutating
+  const isLoading = isLoadingEntity || isLoadingColumnModels
 
   // TODO: the hook above is not returning a stable reference. this is unexpected.
   const queryResultBundle = useDeepCompareMemoize(_queryResultBundle)
   const { accessToken } = useSynapseContext()
   const onSubmit = useCallback(
-    async (formData: ColumnModelFormData[]) => {
+    (formData: ColumnModelFormData[]) => {
       // Transform the form data into ColumnModels
-      transformFormDataToColumnModels(formData)
+      const columnModels = transformFormDataToColumnModels(formData)
 
-      // Create the request to update the table
-      // This call will create new column models as appropriate
-      const request = await createTableUpdateTransactionRequest(
-        accessToken!,
-        entityId,
-        queryResultBundle!.responseBody!.columnModels!,
-        formData,
-      )
```

Comment:
This step is now included in the call to `mutate`. This is now more simple, and state (loading, error, etc) related to this action is all handled by one `useMutation` hook.